### PR TITLE
164: a follow-up PR of PR  #164 hook up data point graphing to the dal

### LINKED ISSUE #164: Hook up the data-point reading to the DAL
Right now our data points are still statically declared. We ned to hook that up to read data from the DAL

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #164 

# What was the issue
Some clean-up tasks after the main PR in 164 is merged:
- Delete unused files from the spike
- Change @preconds to @pre-conditions
- Refactor DAL's storeCsv method and add more tests
- Add missing documents

### CODE REVIEW

File: my-webxr-app/tests/data/DataLayer.test.tsx
Comment (Aesha7) [inline my-webxr-app/tests/data/DataLayer.test.tsx:245]: Great tests!

Do we need to test if batch data stream entires is null?

File: my-webxr-app/src/data/DataLayer.tsx
Comment (Aesha7) [inline my-webxr-app/src/data/DataLayer.tsx:95]: This seems good to me

File: my-webxr-app/src/repository/DbRepository.tsx
Comment (Aesha7) [inline my-webxr-app/src/repository/DbRepository.tsx:208]: Great clean up with documentation 🙌

**Review by Aesha7 — APPROVED** [review]

File: my-webxr-app/tests/data/DataLayer.test.tsx
Comment (trangnguyen3010) [reply my-webxr-app/tests/data/DataLayer.test.tsx:245]: I don't think we need to since in Typescript. Null and Undefined aren't a value of an object but are rather a type. Since we already restrict the type of argument to BatchedDataStream (which can only be wither Array[string] or Array[number], we're already sure that the variable must be of type BatchedDataStream (not any other type like Null or Undefined) ?

Comment (LongQuanNguyen) [reply my-webxr-app/tests/data/DataLayer.test.tsx:245]: TypeScript is weird ngl. I love the idea, but still hate it. As Trang's comment, we don't need to check it, given TypeScript does it job

File: my-webxr-app/src/data/DataLayer.tsx
Comment (trangnguyen3010) [inline my-webxr-app/src/data/DataLayer.tsx:104]: would this cause too much of a merge conflict with your PR @Jesster2829 @LongQuanNguyen ? if it does, I'll revert it

Comment (LongQuanNguyen) [reply my-webxr-app/src/data/DataLayer.tsx:104]: It may, I am not sure how much it will be. May be wait until Jesse's PR get merged in first

**Comment by trangnguyen3010** [comment]
Not important and too much merge conflict to handle

**Pull request closed without merging**
